app/controllers/api/v1/soap_api_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Api::V1::SoapApiController < Api::V1::BaseController
+  soap_service namespace: 'urn:WashOut', wsse_auth_callback: ->(email, password) {
+    user = User.find_by(email: email)
+    return false unless user.present?
+    return user.valid_password?(password)
+  }
+
+  soap_action :get_final_cites_certificate,
+              args: {
+                CertificateNumber: :string,
+                TokenId: :string,
+                IsoCountryCode: :string
+              },
+              return: :string
+  def get_final_cites_certificate
+    render xml: Adapters::SimpleAdapter.run(Adapter.first).to_xml
```

Comment:
As a side note, we'll need to think how to handle requests for an adapter that doesn't exist. It seems that the SOAP style of doing things is to return a SOAP Fault object with details of error (rather than just a status code). We'll handle that separately, just a heads up that we'll need to think of this and other error conditions and document them.